{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MSS: Get WT scores for a bunch of models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "pd.set_option(\"display.max_columns\", 100)\n",
    "sys.path.append(\"..\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inputs:\n",
    "dms_reference_file = \"../reference_files/DMS_substitutions.csv\"\n",
    "scores_file = \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/benchmarks/DMS_zero_shot/substitutions/Spearman/DMS_substitutions_Spearman_DMS_level.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DMS_id</th>\n",
       "      <th>DMS_filename</th>\n",
       "      <th>UniProt_ID</th>\n",
       "      <th>taxon</th>\n",
       "      <th>source_organism</th>\n",
       "      <th>target_seq</th>\n",
       "      <th>seq_len</th>\n",
       "      <th>includes_multiple_mutants</th>\n",
       "      <th>DMS_total_number_mutants</th>\n",
       "      <th>DMS_number_single_mutants</th>\n",
       "      <th>DMS_number_multiple_mutants</th>\n",
       "      <th>DMS_binarization_cutoff</th>\n",
       "      <th>DMS_binarization_method</th>\n",
       "      <th>first_author</th>\n",
       "      <th>title</th>\n",
       "      <th>year</th>\n",
       "      <th>jo</th>\n",
       "      <th>region_mutated</th>\n",
       "      <th>molecule_name</th>\n",
       "      <th>selection_assay</th>\n",
       "      <th>selection_type</th>\n",
       "      <th>MSA_filename</th>\n",
       "      <th>MSA_start</th>\n",
       "      <th>MSA_end</th>\n",
       "      <th>MSA_len</th>\n",
       "      <th>MSA_bitscore</th>\n",
       "      <th>MSA_theta</th>\n",
       "      <th>MSA_num_seqs</th>\n",
       "      <th>MSA_perc_cov</th>\n",
       "      <th>MSA_num_cov</th>\n",
       "      <th>MSA_N_eff</th>\n",
       "      <th>MSA_Neff_L</th>\n",
       "      <th>MSA_Neff_L_category</th>\n",
       "      <th>MSA_num_significant</th>\n",
       "      <th>MSA_num_significant_L</th>\n",
       "      <th>raw_DMS_filename</th>\n",
       "      <th>raw_DMS_phenotype_name</th>\n",
       "      <th>raw_DMS_directionality</th>\n",
       "      <th>raw_DMS_mutant_column</th>\n",
       "      <th>weight_file_name</th>\n",
       "      <th>pdb_file</th>\n",
       "      <th>pdb_range</th>\n",
       "      <th>ProteinGym_version</th>\n",
       "      <th>raw_mut_offset</th>\n",
       "      <th>coarse_selection_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A140D2T1_ZIKV_Sourisseau_2019</td>\n",
       "      <td>A0A140D2T1_ZIKV_Sourisseau_2019.csv</td>\n",
       "      <td>A0A140D2T1_ZIKV</td>\n",
       "      <td>Virus</td>\n",
       "      <td>Zika virus</td>\n",
       "      <td>MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...</td>\n",
       "      <td>3423</td>\n",
       "      <td>False</td>\n",
       "      <td>9576</td>\n",
       "      <td>9576</td>\n",
       "      <td>0</td>\n",
       "      <td>0.043249</td>\n",
       "      <td>median</td>\n",
       "      <td>Sourisseau</td>\n",
       "      <td>Deep Mutational Scanning Comprehensively Maps ...</td>\n",
       "      <td>2019</td>\n",
       "      <td>10.1128/JVI.01291-19</td>\n",
       "      <td>291-794</td>\n",
       "      <td>Zika virus env</td>\n",
       "      <td>Viral replication</td>\n",
       "      <td>Growth</td>\n",
       "      <td>A0A140D2T1_ZIKV_theta0.99_281-804_11-26-2021_b...</td>\n",
       "      <td>281</td>\n",
       "      <td>804</td>\n",
       "      <td>524</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.01</td>\n",
       "      <td>16501</td>\n",
       "      <td>0.948</td>\n",
       "      <td>497.0</td>\n",
       "      <td>1357.9</td>\n",
       "      <td>2.732193</td>\n",
       "      <td>medium</td>\n",
       "      <td>329.0</td>\n",
       "      <td>0.661972</td>\n",
       "      <td>A0A140D2T1_ZIKV_Sourisseau_growth_2019.csv</td>\n",
       "      <td>effect</td>\n",
       "      <td>1</td>\n",
       "      <td>mutant</td>\n",
       "      <td>A0A140D2T1_ZIKV_theta_0.01.npy</td>\n",
       "      <td>A0A140D2T1_ZIKV.pdb</td>\n",
       "      <td>291-794</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrganismalFitness</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A192B1T2_9HIV1_Haddox_2018</td>\n",
       "      <td>A0A192B1T2_9HIV1_Haddox_2018.csv</td>\n",
       "      <td>A0A192B1T2_9HIV1</td>\n",
       "      <td>Virus</td>\n",
       "      <td>HIV</td>\n",
       "      <td>MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...</td>\n",
       "      <td>852</td>\n",
       "      <td>False</td>\n",
       "      <td>12577</td>\n",
       "      <td>12577</td>\n",
       "      <td>0</td>\n",
       "      <td>-2.200000</td>\n",
       "      <td>manual</td>\n",
       "      <td>Haddox</td>\n",
       "      <td>Mapping mutational effects along the evolution...</td>\n",
       "      <td>2018</td>\n",
       "      <td>10.7554/eLife.34420</td>\n",
       "      <td>30-691</td>\n",
       "      <td>HIV env (BF520)</td>\n",
       "      <td>Viral replication</td>\n",
       "      <td>Growth</td>\n",
       "      <td>A0A192B1T2_9HIV1_theta0.99_full_11-26-2021_b09...</td>\n",
       "      <td>1</td>\n",
       "      <td>852</td>\n",
       "      <td>852</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.01</td>\n",
       "      <td>74854</td>\n",
       "      <td>0.986</td>\n",
       "      <td>840.0</td>\n",
       "      <td>36319.9</td>\n",
       "      <td>43.237976</td>\n",
       "      <td>medium</td>\n",
       "      <td>2382.0</td>\n",
       "      <td>2.835714</td>\n",
       "      <td>A0A192B1T2_9HIV1_Haddox_2018.csv</td>\n",
       "      <td>fitness</td>\n",
       "      <td>1</td>\n",
       "      <td>mutant</td>\n",
       "      <td>A0A192B1T2_9HIV1_theta_0.01.npy</td>\n",
       "      <td>A0A192B1T2_9HIV1.pdb</td>\n",
       "      <td>1-852</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrganismalFitness</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1I9GEU1_NEIME_Kennouche_2019</td>\n",
       "      <td>A0A1I9GEU1_NEIME_Kennouche_2019.csv</td>\n",
       "      <td>A0A1I9GEU1_NEIME</td>\n",
       "      <td>Prokaryote</td>\n",
       "      <td>Neisseria meningitidis</td>\n",
       "      <td>FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...</td>\n",
       "      <td>161</td>\n",
       "      <td>False</td>\n",
       "      <td>922</td>\n",
       "      <td>922</td>\n",
       "      <td>0</td>\n",
       "      <td>0.141000</td>\n",
       "      <td>median</td>\n",
       "      <td>Kennouche</td>\n",
       "      <td>Deep mutational scanning of the Neisseria meni...</td>\n",
       "      <td>2019</td>\n",
       "      <td>10.15252/embj.2019102145</td>\n",
       "      <td>1-161</td>\n",
       "      <td>pilin (PilE)</td>\n",
       "      <td>piliation (20D9 anti-pilus monoclonal Ab), agg...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A0A1I9GEU1_NEIME_full_11-26-2021_b08.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>161</td>\n",
       "      <td>161</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.20</td>\n",
       "      <td>5553</td>\n",
       "      <td>0.857</td>\n",
       "      <td>138.0</td>\n",
       "      <td>2183.6</td>\n",
       "      <td>15.823188</td>\n",
       "      <td>medium</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.521739</td>\n",
       "      <td>A0A1I9GEU1_NEIME_Kennouche_2019.csv</td>\n",
       "      <td>piliation_log2_ratio</td>\n",
       "      <td>1</td>\n",
       "      <td>mutants</td>\n",
       "      <td>A0A1I9GEU1_NEIME_theta_0.2.npy</td>\n",
       "      <td>A0A1I9GEU1_NEIME.pdb</td>\n",
       "      <td>1-161</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A247D711_LISMN_Stadelmann_2021</td>\n",
       "      <td>A0A247D711_LISMN_Stadelmann_2021.csv</td>\n",
       "      <td>A0A247D711_LISMN</td>\n",
       "      <td>Eukaryote</td>\n",
       "      <td>Listeria monocytogenes</td>\n",
       "      <td>MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...</td>\n",
       "      <td>87</td>\n",
       "      <td>False</td>\n",
       "      <td>1653</td>\n",
       "      <td>1653</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.015563</td>\n",
       "      <td>median</td>\n",
       "      <td>Stadelmann</td>\n",
       "      <td>A deep mutational scanning platform to charact...</td>\n",
       "      <td>2021</td>\n",
       "      <td>10.1101/2021.08.21.457204</td>\n",
       "      <td>1-87</td>\n",
       "      <td>Anti-CRISPR protein AcrIIA4</td>\n",
       "      <td>activity against SpyCas9 inducing an RFP reporter</td>\n",
       "      <td>Flow cytometry</td>\n",
       "      <td>A0A247D711_LISMN_full_b0.3.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>87</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.20</td>\n",
       "      <td>1316890</td>\n",
       "      <td>1.000</td>\n",
       "      <td>87.0</td>\n",
       "      <td>188739.9</td>\n",
       "      <td>2169.424138</td>\n",
       "      <td>High</td>\n",
       "      <td>209.0</td>\n",
       "      <td>2.402299</td>\n",
       "      <td>A0A247D711_LISMN_Stadelmann_2021.csv</td>\n",
       "      <td>mean_prediction</td>\n",
       "      <td>1</td>\n",
       "      <td>mutant</td>\n",
       "      <td>A0A247D711_LISMN_b03_theta_0.2.npy</td>\n",
       "      <td>A0A247D711_LISMN.pdb</td>\n",
       "      <td>1-87</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A2Z5U3Z0_9INFA_Doud_2016</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA_Doud_2016.csv</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA</td>\n",
       "      <td>Virus</td>\n",
       "      <td>influenza H1N1</td>\n",
       "      <td>MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...</td>\n",
       "      <td>565</td>\n",
       "      <td>False</td>\n",
       "      <td>10715</td>\n",
       "      <td>10715</td>\n",
       "      <td>0</td>\n",
       "      <td>-2.239943</td>\n",
       "      <td>median</td>\n",
       "      <td>Doud</td>\n",
       "      <td>Accurate Measurement of the Effects of All Ami...</td>\n",
       "      <td>2016</td>\n",
       "      <td>10.3390/v8060155</td>\n",
       "      <td>2-565</td>\n",
       "      <td>Influenza hemagglutinin</td>\n",
       "      <td>viral replication</td>\n",
       "      <td>Growth</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA_theta0.99_full_11-26-2021_b09...</td>\n",
       "      <td>1</td>\n",
       "      <td>565</td>\n",
       "      <td>565</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.01</td>\n",
       "      <td>57581</td>\n",
       "      <td>0.968</td>\n",
       "      <td>547.0</td>\n",
       "      <td>9809.4</td>\n",
       "      <td>17.933090</td>\n",
       "      <td>medium</td>\n",
       "      <td>925.0</td>\n",
       "      <td>1.691042</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA_Doud_2016.csv</td>\n",
       "      <td>transformed_pref</td>\n",
       "      <td>1</td>\n",
       "      <td>mutant</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA_theta_0.01.npy</td>\n",
       "      <td>A0A2Z5U3Z0_9INFA.pdb</td>\n",
       "      <td>1-565</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrganismalFitness</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>VKOR1_HUMAN_Chiasson_2020_activity</td>\n",
       "      <td>VKOR1_HUMAN_Chiasson_2020_activity.csv</td>\n",
       "      <td>VKOR1_HUMAN</td>\n",
       "      <td>Human</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...</td>\n",
       "      <td>163</td>\n",
       "      <td>False</td>\n",
       "      <td>697</td>\n",
       "      <td>697</td>\n",
       "      <td>0</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>manual</td>\n",
       "      <td>Chiasson</td>\n",
       "      <td>Multiplexed measurement of variant abundance a...</td>\n",
       "      <td>2020</td>\n",
       "      <td>10.7554/eLife.58026</td>\n",
       "      <td>3-163</td>\n",
       "      <td>VKORC1</td>\n",
       "      <td>carboxylation activity (carboxylation reporter...</td>\n",
       "      <td>enzymatic activity</td>\n",
       "      <td>VKOR1_HUMAN_full_11-26-2021_b03.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>163</td>\n",
       "      <td>163</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.20</td>\n",
       "      <td>14510</td>\n",
       "      <td>0.779</td>\n",
       "      <td>127.0</td>\n",
       "      <td>4655.0</td>\n",
       "      <td>36.653543</td>\n",
       "      <td>medium</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.763780</td>\n",
       "      <td>VKOR1_HUMAN_Chiasson_2020.csv</td>\n",
       "      <td>activity_score</td>\n",
       "      <td>1</td>\n",
       "      <td>variant</td>\n",
       "      <td>VKOR1_HUMAN_theta_0.2.npy</td>\n",
       "      <td>VKOR1_HUMAN.pdb</td>\n",
       "      <td>1-163</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>VRPI_BPT7_Tsuboyama_2023_2WNM</td>\n",
       "      <td>VRPI_BPT7_Tsuboyama_2023_2WNM.csv</td>\n",
       "      <td>VRPI_BPT7</td>\n",
       "      <td>Virus</td>\n",
       "      <td>Escherichia phage</td>\n",
       "      <td>SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>1047</td>\n",
       "      <td>1047</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.100000</td>\n",
       "      <td>manual</td>\n",
       "      <td>Tsuboyama</td>\n",
       "      <td>Mega-scale experimental analysis of protein fo...</td>\n",
       "      <td>2023</td>\n",
       "      <td>10.1038/s41586-023-06328-67</td>\n",
       "      <td>1-56</td>\n",
       "      <td>Bacterial RNA polymerase inhibitor</td>\n",
       "      <td>Stability</td>\n",
       "      <td>cDNA display proteolysis</td>\n",
       "      <td>VRPI_BPT7_2023-08-07_b02.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.01</td>\n",
       "      <td>6266</td>\n",
       "      <td>0.875</td>\n",
       "      <td>49.0</td>\n",
       "      <td>1555.8</td>\n",
       "      <td>31.751020</td>\n",
       "      <td>Medium</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.061224</td>\n",
       "      <td>Tsuboyama2023_Dataset2_Dataset64</td>\n",
       "      <td>ddG_ML_float</td>\n",
       "      <td>1</td>\n",
       "      <td>mut_type</td>\n",
       "      <td>VRPI_BPT7_theta0.01_2023-08-07_b02.npy</td>\n",
       "      <td>VRPI_BPT7.pdb</td>\n",
       "      <td>1-56</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stability</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>YAIA_ECOLI_Tsuboyama_2023_2KVT</td>\n",
       "      <td>YAIA_ECOLI_Tsuboyama_2023_2KVT.csv</td>\n",
       "      <td>YAIA_ECOLI</td>\n",
       "      <td>Prokaryote</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...</td>\n",
       "      <td>52</td>\n",
       "      <td>True</td>\n",
       "      <td>1890</td>\n",
       "      <td>928</td>\n",
       "      <td>962</td>\n",
       "      <td>-1.953132</td>\n",
       "      <td>median</td>\n",
       "      <td>Tsuboyama</td>\n",
       "      <td>Mega-scale experimental analysis of protein fo...</td>\n",
       "      <td>2023</td>\n",
       "      <td>10.1038/s41586-023-06328-68</td>\n",
       "      <td>1-52</td>\n",
       "      <td>Uncharacterized protein YaiA</td>\n",
       "      <td>Stability</td>\n",
       "      <td>cDNA display proteolysis</td>\n",
       "      <td>YAIA_ECOLI_2023-08-07_b03.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.20</td>\n",
       "      <td>5877</td>\n",
       "      <td>0.788</td>\n",
       "      <td>41.0</td>\n",
       "      <td>737.2</td>\n",
       "      <td>17.980488</td>\n",
       "      <td>Medium</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.121951</td>\n",
       "      <td>Tsuboyama2023_Dataset2_Dataset65</td>\n",
       "      <td>ddG_ML_float</td>\n",
       "      <td>1</td>\n",
       "      <td>mut_type</td>\n",
       "      <td>YAIA_ECOLI_theta0.2_2023-08-07_b03.npy</td>\n",
       "      <td>YAIA_ECOLI.pdb</td>\n",
       "      <td>1-52</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stability</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>YAP1_HUMAN_Araya_2012</td>\n",
       "      <td>YAP1_HUMAN_Araya_2012.csv</td>\n",
       "      <td>YAP1_HUMAN</td>\n",
       "      <td>Human</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...</td>\n",
       "      <td>504</td>\n",
       "      <td>True</td>\n",
       "      <td>10075</td>\n",
       "      <td>362</td>\n",
       "      <td>9713</td>\n",
       "      <td>0.623640</td>\n",
       "      <td>median</td>\n",
       "      <td>Araya</td>\n",
       "      <td>A fundamental protein property, thermodynamic ...</td>\n",
       "      <td>2012</td>\n",
       "      <td>10.1073/pnas.1209751109</td>\n",
       "      <td>170-203</td>\n",
       "      <td>YAP1</td>\n",
       "      <td>peptide binding</td>\n",
       "      <td>Binding</td>\n",
       "      <td>YAP1_HUMAN_full_11-26-2021_b02.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>504</td>\n",
       "      <td>504</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.20</td>\n",
       "      <td>1604</td>\n",
       "      <td>0.859</td>\n",
       "      <td>433.0</td>\n",
       "      <td>132.6</td>\n",
       "      <td>0.306236</td>\n",
       "      <td>low</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.002309</td>\n",
       "      <td>YAP1_HUMAN_Araya_2012.csv</td>\n",
       "      <td>W</td>\n",
       "      <td>1</td>\n",
       "      <td>mutant</td>\n",
       "      <td>YAP1_HUMAN_theta_0.2.npy</td>\n",
       "      <td>YAP1_HUMAN.pdb</td>\n",
       "      <td>1-504</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>YNZC_BACSU_Tsuboyama_2023_2JVD</td>\n",
       "      <td>YNZC_BACSU_Tsuboyama_2023_2JVD.csv</td>\n",
       "      <td>YNZC_BACSU</td>\n",
       "      <td>Prokaryote</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK</td>\n",
       "      <td>39</td>\n",
       "      <td>True</td>\n",
       "      <td>2300</td>\n",
       "      <td>714</td>\n",
       "      <td>1586</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>manual</td>\n",
       "      <td>Tsuboyama</td>\n",
       "      <td>Mega-scale experimental analysis of protein fo...</td>\n",
       "      <td>2023</td>\n",
       "      <td>10.1038/s41586-023-06328-69</td>\n",
       "      <td>1-39</td>\n",
       "      <td>UPF0291 protein YnzC</td>\n",
       "      <td>Stability</td>\n",
       "      <td>cDNA display proteolysis</td>\n",
       "      <td>YNZC_BACSU_2023-08-07_b07.a2m</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.20</td>\n",
       "      <td>7116</td>\n",
       "      <td>0.974</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1588.3</td>\n",
       "      <td>41.797368</td>\n",
       "      <td>Medium</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.342105</td>\n",
       "      <td>Tsuboyama2023_Dataset2_Dataset66</td>\n",
       "      <td>ddG_ML_float</td>\n",
       "      <td>1</td>\n",
       "      <td>mut_type</td>\n",
       "      <td>YNZC_BACSU_theta0.2_2023-08-07_b07.npy</td>\n",
       "      <td>YNZC_BACSU.pdb</td>\n",
       "      <td>1-39</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stability</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 DMS_id  \\\n",
       "0       A0A140D2T1_ZIKV_Sourisseau_2019   \n",
       "1          A0A192B1T2_9HIV1_Haddox_2018   \n",
       "2       A0A1I9GEU1_NEIME_Kennouche_2019   \n",
       "3      A0A247D711_LISMN_Stadelmann_2021   \n",
       "4            A0A2Z5U3Z0_9INFA_Doud_2016   \n",
       "..                                  ...   \n",
       "212  VKOR1_HUMAN_Chiasson_2020_activity   \n",
       "213       VRPI_BPT7_Tsuboyama_2023_2WNM   \n",
       "214      YAIA_ECOLI_Tsuboyama_2023_2KVT   \n",
       "215               YAP1_HUMAN_Araya_2012   \n",
       "216      YNZC_BACSU_Tsuboyama_2023_2JVD   \n",
       "\n",
       "                               DMS_filename        UniProt_ID       taxon  \\\n",
       "0       A0A140D2T1_ZIKV_Sourisseau_2019.csv   A0A140D2T1_ZIKV       Virus   \n",
       "1          A0A192B1T2_9HIV1_Haddox_2018.csv  A0A192B1T2_9HIV1       Virus   \n",
       "2       A0A1I9GEU1_NEIME_Kennouche_2019.csv  A0A1I9GEU1_NEIME  Prokaryote   \n",
       "3      A0A247D711_LISMN_Stadelmann_2021.csv  A0A247D711_LISMN   Eukaryote   \n",
       "4            A0A2Z5U3Z0_9INFA_Doud_2016.csv  A0A2Z5U3Z0_9INFA       Virus   \n",
       "..                                      ...               ...         ...   \n",
       "212  VKOR1_HUMAN_Chiasson_2020_activity.csv       VKOR1_HUMAN       Human   \n",
       "213       VRPI_BPT7_Tsuboyama_2023_2WNM.csv         VRPI_BPT7       Virus   \n",
       "214      YAIA_ECOLI_Tsuboyama_2023_2KVT.csv        YAIA_ECOLI  Prokaryote   \n",
       "215               YAP1_HUMAN_Araya_2012.csv        YAP1_HUMAN       Human   \n",
       "216      YNZC_BACSU_Tsuboyama_2023_2JVD.csv        YNZC_BACSU  Prokaryote   \n",
       "\n",
       "            source_organism  \\\n",
       "0                Zika virus   \n",
       "1                       HIV   \n",
       "2    Neisseria meningitidis   \n",
       "3    Listeria monocytogenes   \n",
       "4            influenza H1N1   \n",
       "..                      ...   \n",
       "212            Homo sapiens   \n",
       "213       Escherichia phage   \n",
       "214        Escherichia coli   \n",
       "215            Homo sapiens   \n",
       "216       Bacillus subtilis   \n",
       "\n",
       "                                            target_seq  seq_len  \\\n",
       "0    MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...     3423   \n",
       "1    MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...      852   \n",
       "2    FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...      161   \n",
       "3    MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...       87   \n",
       "4    MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...      565   \n",
       "..                                                 ...      ...   \n",
       "212  MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...      163   \n",
       "213  SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...       56   \n",
       "214  PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...       52   \n",
       "215  MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...      504   \n",
       "216            MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK       39   \n",
       "\n",
       "     includes_multiple_mutants  DMS_total_number_mutants  \\\n",
       "0                        False                      9576   \n",
       "1                        False                     12577   \n",
       "2                        False                       922   \n",
       "3                        False                      1653   \n",
       "4                        False                     10715   \n",
       "..                         ...                       ...   \n",
       "212                      False                       697   \n",
       "213                      False                      1047   \n",
       "214                       True                      1890   \n",
       "215                       True                     10075   \n",
       "216                       True                      2300   \n",
       "\n",
       "     DMS_number_single_mutants  DMS_number_multiple_mutants  \\\n",
       "0                         9576                            0   \n",
       "1                        12577                            0   \n",
       "2                          922                            0   \n",
       "3                         1653                            0   \n",
       "4                        10715                            0   \n",
       "..                         ...                          ...   \n",
       "212                        697                            0   \n",
       "213                       1047                            0   \n",
       "214                        928                          962   \n",
       "215                        362                         9713   \n",
       "216                        714                         1586   \n",
       "\n",
       "     DMS_binarization_cutoff DMS_binarization_method first_author  \\\n",
       "0                   0.043249                  median   Sourisseau   \n",
       "1                  -2.200000                  manual       Haddox   \n",
       "2                   0.141000                  median    Kennouche   \n",
       "3                  -0.015563                  median   Stadelmann   \n",
       "4                  -2.239943                  median         Doud   \n",
       "..                       ...                     ...          ...   \n",
       "212                 0.700000                  manual     Chiasson   \n",
       "213                -1.100000                  manual    Tsuboyama   \n",
       "214                -1.953132                  median    Tsuboyama   \n",
       "215                 0.623640                  median        Araya   \n",
       "216                -1.000000                  manual    Tsuboyama   \n",
       "\n",
       "                                                 title  year  \\\n",
       "0    Deep Mutational Scanning Comprehensively Maps ...  2019   \n",
       "1    Mapping mutational effects along the evolution...  2018   \n",
       "2    Deep mutational scanning of the Neisseria meni...  2019   \n",
       "3    A deep mutational scanning platform to charact...  2021   \n",
       "4    Accurate Measurement of the Effects of All Ami...  2016   \n",
       "..                                                 ...   ...   \n",
       "212  Multiplexed measurement of variant abundance a...  2020   \n",
       "213  Mega-scale experimental analysis of protein fo...  2023   \n",
       "214  Mega-scale experimental analysis of protein fo...  2023   \n",
       "215  A fundamental protein property, thermodynamic ...  2012   \n",
       "216  Mega-scale experimental analysis of protein fo...  2023   \n",
       "\n",
       "                              jo region_mutated  \\\n",
       "0           10.1128/JVI.01291-19        291-794   \n",
       "1            10.7554/eLife.34420         30-691   \n",
       "2       10.15252/embj.2019102145          1-161   \n",
       "3      10.1101/2021.08.21.457204           1-87   \n",
       "4               10.3390/v8060155          2-565   \n",
       "..                           ...            ...   \n",
       "212          10.7554/eLife.58026          3-163   \n",
       "213  10.1038/s41586-023-06328-67           1-56   \n",
       "214  10.1038/s41586-023-06328-68           1-52   \n",
       "215      10.1073/pnas.1209751109        170-203   \n",
       "216  10.1038/s41586-023-06328-69           1-39   \n",
       "\n",
       "                          molecule_name  \\\n",
       "0                        Zika virus env   \n",
       "1                       HIV env (BF520)   \n",
       "2                          pilin (PilE)   \n",
       "3           Anti-CRISPR protein AcrIIA4   \n",
       "4               Influenza hemagglutinin   \n",
       "..                                  ...   \n",
       "212                              VKORC1   \n",
       "213  Bacterial RNA polymerase inhibitor   \n",
       "214        Uncharacterized protein YaiA   \n",
       "215                                YAP1   \n",
       "216                UPF0291 protein YnzC   \n",
       "\n",
       "                                       selection_assay  \\\n",
       "0                                    Viral replication   \n",
       "1                                    Viral replication   \n",
       "2    piliation (20D9 anti-pilus monoclonal Ab), agg...   \n",
       "3    activity against SpyCas9 inducing an RFP reporter   \n",
       "4                                    viral replication   \n",
       "..                                                 ...   \n",
       "212  carboxylation activity (carboxylation reporter...   \n",
       "213                                          Stability   \n",
       "214                                          Stability   \n",
       "215                                    peptide binding   \n",
       "216                                          Stability   \n",
       "\n",
       "               selection_type  \\\n",
       "0                      Growth   \n",
       "1                      Growth   \n",
       "2                         NaN   \n",
       "3              Flow cytometry   \n",
       "4                      Growth   \n",
       "..                        ...   \n",
       "212        enzymatic activity   \n",
       "213  cDNA display proteolysis   \n",
       "214  cDNA display proteolysis   \n",
       "215                   Binding   \n",
       "216  cDNA display proteolysis   \n",
       "\n",
       "                                          MSA_filename  MSA_start  MSA_end  \\\n",
       "0    A0A140D2T1_ZIKV_theta0.99_281-804_11-26-2021_b...        281      804   \n",
       "1    A0A192B1T2_9HIV1_theta0.99_full_11-26-2021_b09...          1      852   \n",
       "2             A0A1I9GEU1_NEIME_full_11-26-2021_b08.a2m          1      161   \n",
       "3                       A0A247D711_LISMN_full_b0.3.a2m          1       87   \n",
       "4    A0A2Z5U3Z0_9INFA_theta0.99_full_11-26-2021_b09...          1      565   \n",
       "..                                                 ...        ...      ...   \n",
       "212                VKOR1_HUMAN_full_11-26-2021_b03.a2m          1      163   \n",
       "213                       VRPI_BPT7_2023-08-07_b02.a2m          1       56   \n",
       "214                      YAIA_ECOLI_2023-08-07_b03.a2m          1       52   \n",
       "215                 YAP1_HUMAN_full_11-26-2021_b02.a2m          1      504   \n",
       "216                      YNZC_BACSU_2023-08-07_b07.a2m          1       39   \n",
       "\n",
       "     MSA_len  MSA_bitscore  MSA_theta  MSA_num_seqs  MSA_perc_cov  \\\n",
       "0        524           0.2       0.01         16501         0.948   \n",
       "1        852           0.9       0.01         74854         0.986   \n",
       "2        161           0.8       0.20          5553         0.857   \n",
       "3         87           0.2       0.20       1316890         1.000   \n",
       "4        565           0.9       0.01         57581         0.968   \n",
       "..       ...           ...        ...           ...           ...   \n",
       "212      163           0.3       0.20         14510         0.779   \n",
       "213       56           0.2       0.01          6266         0.875   \n",
       "214       52           0.3       0.20          5877         0.788   \n",
       "215      504           0.2       0.20          1604         0.859   \n",
       "216       39           0.7       0.20          7116         0.974   \n",
       "\n",
       "     MSA_num_cov  MSA_N_eff   MSA_Neff_L MSA_Neff_L_category  \\\n",
       "0          497.0     1357.9     2.732193              medium   \n",
       "1          840.0    36319.9    43.237976              medium   \n",
       "2          138.0     2183.6    15.823188              medium   \n",
       "3           87.0   188739.9  2169.424138                High   \n",
       "4          547.0     9809.4    17.933090              medium   \n",
       "..           ...        ...          ...                 ...   \n",
       "212        127.0     4655.0    36.653543              medium   \n",
       "213         49.0     1555.8    31.751020              Medium   \n",
       "214         41.0      737.2    17.980488              Medium   \n",
       "215        433.0      132.6     0.306236                 low   \n",
       "216         38.0     1588.3    41.797368              Medium   \n",
       "\n",
       "     MSA_num_significant  MSA_num_significant_L  \\\n",
       "0                  329.0               0.661972   \n",
       "1                 2382.0               2.835714   \n",
       "2                   72.0               0.521739   \n",
       "3                  209.0               2.402299   \n",
       "4                  925.0               1.691042   \n",
       "..                   ...                    ...   \n",
       "212                 97.0               0.763780   \n",
       "213                  3.0               0.061224   \n",
       "214                  5.0               0.121951   \n",
       "215                  1.0               0.002309   \n",
       "216                 13.0               0.342105   \n",
       "\n",
       "                               raw_DMS_filename raw_DMS_phenotype_name  \\\n",
       "0    A0A140D2T1_ZIKV_Sourisseau_growth_2019.csv                 effect   \n",
       "1              A0A192B1T2_9HIV1_Haddox_2018.csv                fitness   \n",
       "2           A0A1I9GEU1_NEIME_Kennouche_2019.csv   piliation_log2_ratio   \n",
       "3          A0A247D711_LISMN_Stadelmann_2021.csv        mean_prediction   \n",
       "4                A0A2Z5U3Z0_9INFA_Doud_2016.csv       transformed_pref   \n",
       "..                                          ...                    ...   \n",
       "212               VKOR1_HUMAN_Chiasson_2020.csv         activity_score   \n",
       "213            Tsuboyama2023_Dataset2_Dataset64           ddG_ML_float   \n",
       "214            Tsuboyama2023_Dataset2_Dataset65           ddG_ML_float   \n",
       "215                   YAP1_HUMAN_Araya_2012.csv                      W   \n",
       "216            Tsuboyama2023_Dataset2_Dataset66           ddG_ML_float   \n",
       "\n",
       "     raw_DMS_directionality raw_DMS_mutant_column  \\\n",
       "0                         1                mutant   \n",
       "1                         1                mutant   \n",
       "2                         1               mutants   \n",
       "3                         1                mutant   \n",
       "4                         1                mutant   \n",
       "..                      ...                   ...   \n",
       "212                       1               variant   \n",
       "213                       1              mut_type   \n",
       "214                       1              mut_type   \n",
       "215                       1                mutant   \n",
       "216                       1              mut_type   \n",
       "\n",
       "                           weight_file_name              pdb_file pdb_range  \\\n",
       "0            A0A140D2T1_ZIKV_theta_0.01.npy   A0A140D2T1_ZIKV.pdb   291-794   \n",
       "1           A0A192B1T2_9HIV1_theta_0.01.npy  A0A192B1T2_9HIV1.pdb     1-852   \n",
       "2            A0A1I9GEU1_NEIME_theta_0.2.npy  A0A1I9GEU1_NEIME.pdb     1-161   \n",
       "3        A0A247D711_LISMN_b03_theta_0.2.npy  A0A247D711_LISMN.pdb      1-87   \n",
       "4           A0A2Z5U3Z0_9INFA_theta_0.01.npy  A0A2Z5U3Z0_9INFA.pdb     1-565   \n",
       "..                                      ...                   ...       ...   \n",
       "212               VKOR1_HUMAN_theta_0.2.npy       VKOR1_HUMAN.pdb     1-163   \n",
       "213  VRPI_BPT7_theta0.01_2023-08-07_b02.npy         VRPI_BPT7.pdb      1-56   \n",
       "214  YAIA_ECOLI_theta0.2_2023-08-07_b03.npy        YAIA_ECOLI.pdb      1-52   \n",
       "215                YAP1_HUMAN_theta_0.2.npy        YAP1_HUMAN.pdb     1-504   \n",
       "216  YNZC_BACSU_theta0.2_2023-08-07_b07.npy        YNZC_BACSU.pdb      1-39   \n",
       "\n",
       "     ProteinGym_version  raw_mut_offset coarse_selection_type  \n",
       "0                   0.1             NaN     OrganismalFitness  \n",
       "1                   0.1             NaN     OrganismalFitness  \n",
       "2                   0.1             NaN              Activity  \n",
       "3                   1.0             NaN              Activity  \n",
       "4                   0.1             NaN     OrganismalFitness  \n",
       "..                  ...             ...                   ...  \n",
       "212                 0.1             NaN              Activity  \n",
       "213                 1.0             NaN             Stability  \n",
       "214                 1.0             NaN             Stability  \n",
       "215                 0.1             NaN               Binding  \n",
       "216                 1.0             NaN             Stability  \n",
       "\n",
       "[217 rows x 45 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the reference file\n",
    "df_ref = pd.read_csv(dms_reference_file)\n",
    "df_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "example_mapping = pd.DataFrame({\"DMS_id\": [\"all_wt\"], \"DMS_filename\": [\"all_wt.csv\"]})  # TODO later this should just point to a CSV / fasta to score all sequences\n",
    "# example_mapping.to_csv(\"all_wt_mapping.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutated_sequence</th>\n",
       "      <th>DMS_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>187 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      mutated_sequence  DMS_score\n",
       "0    MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...          0\n",
       "1    MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...          0\n",
       "2    FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...          0\n",
       "3    MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...          0\n",
       "4    MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...          0\n",
       "..                                                 ...        ...\n",
       "182  MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...          0\n",
       "183  SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...          0\n",
       "184  PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...          0\n",
       "185  MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...          0\n",
       "186            MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK          0\n",
       "\n",
       "[187 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example_csv = df_ref[[\"target_seq\"]].drop_duplicates().reset_index(drop=True).rename(columns={\"target_seq\": \"mutated_sequence\"})\n",
    "example_csv[\"DMS_score\"] = 0\n",
    "# example_csv.to_csv(\"all_wt.csv\", index=False)\n",
    "example_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make per-sequence files\n",
    "example_mapping_per_sequence = df_ref.drop_duplicates(subset=\"target_seq\")[[\"DMS_id\", \"DMS_filename\"]]\n",
    "# example_mapping_per_sequence.to_csv(\"per_wt_mapping.csv\", index=False)\n",
    "for i, row in df_ref.drop_duplicates(subset=\"target_seq\").iterrows():\n",
    "    tmp_df = pd.DataFrame([row[\"target_seq\"]], columns=[\"mutated_sequence\"]).rename(columns={\"target_seq\": \"mutated_sequence\"})\n",
    "    tmp_df[\"DMS_score\"] = 0\n",
    "    # tmp_df.to_csv(\"./per_wt_seq/\" + row[\"DMS_filename\"], index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example 1: Get ProGen scores (start small?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maximum context length: 2048\n",
      "Computing scores for: all_wt with Progen2: ../checkpoints/progen2-base\n",
      "Model: progen2-base\n",
      "100%|█████████████████████████████████████████████| 1/1 [00:00<00:00,  1.17it/s]\n",
      "[-409.02035522] -1.9 [407.12035522]\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/progen2/compute_fitness.py\", line 185, in <module>\n",
      "    main()\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/progen2/compute_fitness.py\", line 172, in main\n",
      "    assert abs(model_scores - checkpoint_x_ll[model_size][1]) < 0.1\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "AssertionError\n"
     ]
    }
   ],
   "source": [
    "# Takes 9 seconds for 2 seqs, 217 sequences would take 15min\n",
    "\n",
    "index = 0\n",
    "command = f\"../proteingym/baselines/progen2/compute_fitness.py \\\n",
    "            --Progen2_model_name_or_path ../checkpoints/progen2-base \\\n",
    "            --DMS_reference_file_path all_wt_mapping.csv \\\n",
    "            --DMS_data_folder . \\\n",
    "            --DMS_index {index} \\\n",
    "            --output_scores_folder ./model_scores_out/ \\\n",
    "            --indel_mode \\\n",
    "            --device mps \\\n",
    "            --test \\\n",
    "            --raw_score\"\n",
    "# command = f\"../proteingym/baselines/progen2/likelihood.py --model_ckpt ../checkpoints/progen2-base/  \\\n",
    "#     --context {example_csv.iloc[index]['mutated_sequence']}  \\\n",
    "#     --device mps\"\n",
    "! python {command}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ESM pseudolikelihoods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutated_sequence</th>\n",
       "      <th>DMS_score</th>\n",
       "      <th>esm1v_t33_650M_UR90S_1</th>\n",
       "      <th>Ensemble_ESM1v</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...</td>\n",
       "      <td>0</td>\n",
       "      <td>-12941.488922</td>\n",
       "      <td>-12941.488922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...</td>\n",
       "      <td>0</td>\n",
       "      <td>-7479.261695</td>\n",
       "      <td>-7479.261695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...</td>\n",
       "      <td>0</td>\n",
       "      <td>-1550.662315</td>\n",
       "      <td>-1550.662315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...</td>\n",
       "      <td>0</td>\n",
       "      <td>-354.171648</td>\n",
       "      <td>-354.171648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...</td>\n",
       "      <td>0</td>\n",
       "      <td>-5073.500586</td>\n",
       "      <td>-5073.500586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...</td>\n",
       "      <td>0</td>\n",
       "      <td>-1218.057225</td>\n",
       "      <td>-1218.057225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...</td>\n",
       "      <td>0</td>\n",
       "      <td>-226.470604</td>\n",
       "      <td>-226.470604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...</td>\n",
       "      <td>0</td>\n",
       "      <td>-220.723548</td>\n",
       "      <td>-220.723548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...</td>\n",
       "      <td>0</td>\n",
       "      <td>-4214.837010</td>\n",
       "      <td>-4214.837010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK</td>\n",
       "      <td>0</td>\n",
       "      <td>-228.818912</td>\n",
       "      <td>-228.818912</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>187 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      mutated_sequence  DMS_score  \\\n",
       "0    MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...          0   \n",
       "1    MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...          0   \n",
       "2    FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...          0   \n",
       "3    MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...          0   \n",
       "4    MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...          0   \n",
       "..                                                 ...        ...   \n",
       "182  MGSTWGSPGWVRLALCLTGLVLSLYALHVKAARARDRDYRALCDVG...          0   \n",
       "183  SLSVDNKKFWATVESSEHSFEVPIYAETLDEALELAEWQYVPAGFE...          0   \n",
       "184  PREAYIVTIEKGKPGQTVTWYQLRADHPKPDSLISEHPTAQEAMDA...          0   \n",
       "185  MDPGQQPPPQPAPQGQGQPPSQPPQGQGPPSGPGQPAPAATQAAPQ...          0   \n",
       "186            MISNAKIARINELAAKAKAGVITEEEKAEQQKLRQEYLK          0   \n",
       "\n",
       "     esm1v_t33_650M_UR90S_1  Ensemble_ESM1v  \n",
       "0             -12941.488922   -12941.488922  \n",
       "1              -7479.261695    -7479.261695  \n",
       "2              -1550.662315    -1550.662315  \n",
       "3               -354.171648     -354.171648  \n",
       "4              -5073.500586    -5073.500586  \n",
       "..                      ...             ...  \n",
       "182            -1218.057225    -1218.057225  \n",
       "183             -226.470604     -226.470604  \n",
       "184             -220.723548     -220.723548  \n",
       "185            -4214.837010    -4214.837010  \n",
       "186             -228.818912     -228.818912  \n",
       "\n",
       "[187 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Stitch together output files into all_wt.csv\n",
    "dirs = [\"esm2_t30_150M_UR50D\", \"esm1v_t33_650M_UR90S_1\"]\n",
    "import os\n",
    "for dir in dirs:\n",
    "    dfs = []\n",
    "    dir_path = f\"./model_scores_out/{dir}/\"\n",
    "    for filename in sorted(os.listdir(dir_path)):\n",
    "        if filename == \"all_wt.csv\":\n",
    "            continue\n",
    "        dms_id = filename.split(\".csv\")[0]\n",
    "        tmp = pd.read_csv(dir_path+filename)\n",
    "        tmp[\"DMS_id\"] = dms_id\n",
    "        dfs.append(tmp)\n",
    "    tmp_full = pd.concat(dfs)\n",
    "    model_col = dir\n",
    "    tmp_full[model_col] = tmp_full[model_col].replace(0, np.nan)\n",
    "    tmp_full = tmp_full.drop(columns=[c for c in tmp_full if \"Ensemble\" in c])\n",
    "    tmp_full = tmp_full.drop(columns=[\"DMS_score\"])\n",
    "    tmp_full.to_csv(f\"{dir_path}/all_wt.csv\", index=False)\n",
    "\n",
    "df_tmp = pd.read_csv(\"./model_scores_out/esm1v_t33_650M_UR90S_1_wt_pseudo_ppl/all_wt.csv\")\n",
    "df_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tmp compute using ESM2 endpoint\n",
    "# token = HUGGINGFACE_TOKEN\n",
    "\n",
    "# import requests\n",
    "\n",
    "# API_URL = \"https://api-inference.huggingface.co/models/facebook/esm2_t6_8M_UR50D\"\n",
    "# headers = {\"Authorization\": f\"Bearer {token}\"}\n",
    "\n",
    "# def query(payload):\n",
    "# \tresponse = requests.post(API_URL, headers=headers, json=payload)\n",
    "# \treturn response.json()\n",
    "\t\n",
    "# output = query({\n",
    "# \t\"inputs\": \"MQIFVKTLTGKTITLEVEPS<mask>MIENVKAKIQDKEGIPPDQQRLIFAGKQLEDGRTLSDYNIQKESTLHLVLRLRGG\",\n",
    "# })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Arguments: Namespace(model_type=['ESM1v'], model_location=['esm1v_t33_650M_UR90S_1'], sequence=None, dms_input=PosixPath('.'), dms_index=0, dms_mapping='all_wt_mapping.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm1v_t33_650M_UR90S_1_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=None, msa_sampling_strategy='sequence-reweighting', msa_samples=400, msa_weights_folder=None, seeds=1, filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='optimal', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: all_wt\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...          0\n",
      "1  MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...          0\n",
      "2  FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...          0\n",
      "3  MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...          0\n",
      "4  MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...          0\n",
      "df shape: (187, 2)\n",
      "Starting model scoring\n",
      "/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/baselines/esm/esm/pretrained.py:212: UserWarning: Regression weights not found, predicting contacts will not produce correct results.\n",
      "  warnings.warn(\n",
      "tmp, wt-pseudo-ppl only implemented for mutated_sequence == WT\n",
      "100%|█████████████████████████████████████████| 187/187 [02:47<00:00,  1.11it/s]\n"
     ]
    }
   ],
   "source": [
    "# 144s per sequence, would take 8 hours...\n",
    "# 2 positions per second\n",
    "# esm1v_t33_650M_UR90S_1, but now using much smaller one\n",
    "\n",
    "# Using one file for all the sequences\n",
    "# Using wt-marginals pseudo-ppl (experimental)\n",
    "# command = f\"../proteingym/baselines/esm/compute_fitness.py \\\n",
    "#     --model-location esm1v_t33_650M_UR90S_1 \\\n",
    "#     --model_type ESM1v \\\n",
    "#     --dms_index 0 \\\n",
    "#     --dms_mapping all_wt_mapping.csv \\\n",
    "#     --dms-input . \\\n",
    "#     --dms-output ./model_scores_out/esm1v_t33_650M_UR90S_1_wt_pseudo_ppl/ \\\n",
    "#     --scoring-strategy wt-pseudo-ppl \\\n",
    "#     --scoring-window optimal \\\n",
    "#     --device mps\"  # Should try use mps\n",
    "# ! python {command}\n",
    "\n",
    "# Use per-sequence mapping file\n",
    "# for i in range(0, 188):\n",
    "#     command = f\"../proteingym/baselines/esm/compute_fitness.py \\\n",
    "#         --model-location esm1v_t33_650M_UR90S_1 \\\n",
    "#         --model_type ESM1v \\\n",
    "#         --dms_index {i} \\\n",
    "#         --dms_mapping per_wt_mapping.csv \\\n",
    "#         --dms-input ./per_wt_seq/ \\\n",
    "#         --dms-output ./model_scores_out/esm1v_t33_650M_UR90S_1/ \\\n",
    "#         --scoring-strategy pseudo-ppl \\\n",
    "#         --scoring-window optimal \\\n",
    "#         --device mps\"  # Should try use mps\n",
    "#     ! python {command}\n",
    "\n",
    "# ! echo {1..2} | xargs -n1 -I{} -P 4 python ../proteingym/baselines/esm/compute_fitness.py \\\n",
    "#     --model-location esm2_t30_150M_UR50D \\\n",
    "#     --model_type ESM2 \\\n",
    "#     --dms_index {} \\\n",
    "#     --dms_mapping per_wt_mapping.csv \\\n",
    "#     --dms-input ./per_wt_seq/ \\\n",
    "#     --dms-output ./model_scores_out/esm2_t30_150M_UR50D/ \\\n",
    "#     --scoring-strategy pseudo-ppl \\\n",
    "#     --scoring-window optimal \\\n",
    "#     --device mps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MSAT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=0, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A140D2T1_ZIKV_Sourisseau_2019\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 281 and end 804\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MKNPKKKSGGFRIVNMLKRGVARVNPLGGLKRLPAGLLLGHGPIRM...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Skipping seed 1 as it is already in the dataframe\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=1, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A192B1T2_9HIV1_Haddox_2018\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 852\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MRVKGIQMNSQHLLRWGIMILGMIMICSVAGNLWVTVYYGVPVWKD...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Sampling sequences from MSA with strategy: sequence-reweighting\n",
      "Proportion of sequences dropped due to fraction of gaps: 0.0%\n",
      "Proportion of non-focus columns removed: 0.0%\n",
      "/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py:100: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
      "  self.seq_name_to_sequence[msa_df.index[seq_idx]] = msa_df.sequence[seq_idx]\n",
      "^C\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 616, in <module>\n",
      "    main(args)\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 459, in main\n",
      "    data = [read_msa(filename=args.msa_path, nseq=args.msa_samples, sampling_strategy=args.msa_sampling_strategy, random_seed=seed, weight_filename=MSA_weight_file_name,\n",
      "            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 57, in read_msa\n",
      "    MSA = MSA_processing(\n",
      "          ^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py\", line 52, in __init__\n",
      "    self.gen_alignment()\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py\", line 122, in gen_alignment\n",
      "    self.seq_name_to_sequence[seq_name] = [sequence[ix].upper() for ix in self.focus_cols]\n",
      "                                           ^^^^^^^^^^^^^^^^^^^^\n",
      "KeyboardInterrupt\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=2, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A1I9GEU1_NEIME_Kennouche_2019\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 161\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSA...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Sampling sequences from MSA with strategy: sequence-reweighting\n",
      "Proportion of sequences dropped due to fraction of gaps: 0.0%\n",
      "Proportion of non-focus columns removed: 0.0%\n",
      "/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py:100: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
      "  self.seq_name_to_sequence[msa_df.index[seq_idx]] = msa_df.sequence[seq_idx]\n",
      "Encoding sequences\n",
      "One-hot encoded sequences shape:(5541, 161, 20)\n",
      "Loaded sequence weights from disk\n",
      "Neff = 2265.8900786612826\n",
      "Data Shape = (5541, 161, 20)\n",
      "Neff: 2265.8900786612826\n",
      "Name of focus_seq: >A0A1I9GEU1_NEIME/1-161\n",
      "Check sum weights MSA: 1.0000000000000002\n",
      "First 10 elements of sampled MSA: \n",
      "[('>A0A1I9GEU1_NEIME/1-161', 'FTLIELMIVIAIVGILAAVALPAYQDYTARAQVSEAILLAEGQKSAVTEYYLNHGEWPGDNSSAGVATSADIKGKYVQSVTVANGVITAQMASSNVNNEIKSKKLSLWAKRQNGSVKWFCGQPVTRTTATATDVAAANGKTDDKINTKHLPSTCRDDSSAS'), ('>UniRef100_A0A4U6R2G6/8-146', 'FTLIELMIVVAIIGILAAVAIPAYQDYTARAQVSEGVSLAGGLKSTVSDIWSTKGALDDASNTNGLPAAADVNGSYVNDVSVTDGLITVTMKSSDVSAGIQSATFLLSPITGSGSMIWTCKTGGG-----------------SAMDAKYLPSSCR------')]\n",
      "length of first sequence in MSA: 161\n",
      "Batch sizes: torch.Size([1, 2, 162])\n",
      "tmp MSAT logits shape:  torch.Size([1, 2, 162, 33])\n",
      "token probs before unsqueeze:  torch.Size([1, 162, 33])\n",
      "tmp token probs shape:  torch.Size([1, 162, 33])\n",
      "length between indices: 160\n",
      "Ignoring offset_idx because we've already chopped the WT_sequence to the correct length\n",
      "100%|█████████████████████████████████████| 161/161 [00:00<00:00, 158294.17it/s]\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=3, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A247D711_LISMN_Stadelmann_2021\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 87\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISES...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Sampling sequences from MSA with strategy: sequence-reweighting\n",
      "Proportion of sequences dropped due to fraction of gaps: 0.0%\n",
      "Proportion of non-focus columns removed: 0.0%\n",
      "/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py:100: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
      "  self.seq_name_to_sequence[msa_df.index[seq_idx]] = msa_df.sequence[seq_idx]\n",
      "Encoding sequences\n",
      "One-hot encoded sequences shape:(57, 87, 20)\n",
      "Loaded sequence weights from disk\n",
      "Neff = 8.936227394896125\n",
      "Data Shape = (57, 87, 20)\n",
      "Neff: 8.936227394896125\n",
      "Name of focus_seq: >A0A247D711_LISMN/1-87\n",
      "Check sum weights MSA: 1.0\n",
      "First 10 elements of sampled MSA: \n",
      "[('>A0A247D711_LISMN/1-87', 'MNINDLIREIKNKDYTVKLSGTDSNSITQLIIRVNNDGNEYVISESENESIVEKFISAFKNGWNQEYEDEEEFYNDMQTITLKSELN'), ('>UniRef100_A0A823IUF1/1-83', 'MNLKELVREIKNKDYTAKLSGTDSNSITQLIIHVNNDGNEYGISESNFESIVEKFVSTFENGWDGAYEDEEEFYNDMQDIVNR----')]\n",
      "length of first sequence in MSA: 87\n",
      "Batch sizes: torch.Size([1, 2, 88])\n",
      "tmp MSAT logits shape:  torch.Size([1, 2, 88, 33])\n",
      "token probs before unsqueeze:  torch.Size([1, 88, 33])\n",
      "tmp token probs shape:  torch.Size([1, 88, 33])\n",
      "length between indices: 86\n",
      "Ignoring offset_idx because we've already chopped the WT_sequence to the correct length\n",
      "100%|███████████████████████████████████████| 87/87 [00:00<00:00, 159138.44it/s]\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=4, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A2Z5U3Z0_9INFA_Doud_2016\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 565\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MKAKLLVLLYAFVATDADTICIGYHANNSTDTVDTILEKNVAVTHS...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Sampling sequences from MSA with strategy: sequence-reweighting\n",
      "^C\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 616, in <module>\n",
      "    main(args)\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 459, in main\n",
      "    data = [read_msa(filename=args.msa_path, nseq=args.msa_samples, sampling_strategy=args.msa_sampling_strategy, random_seed=seed, weight_filename=MSA_weight_file_name,\n",
      "            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 57, in read_msa\n",
      "    MSA = MSA_processing(\n",
      "          ^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py\", line 52, in __init__\n",
      "    self.gen_alignment()\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py\", line 78, in gen_alignment\n",
      "    msa_df.sequence = msa_df.sequence.apply(lambda x: x.replace(\".\",\"-\")).apply(lambda x: ''.join([aa.upper() for aa in x]))\n",
      "                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/core/series.py\", line 4924, in apply\n",
      "    ).apply()\n",
      "      ^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/core/apply.py\", line 1427, in apply\n",
      "    return self.apply_standard()\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/core/apply.py\", line 1507, in apply_standard\n",
      "    mapped = obj._map_values(\n",
      "             ^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/core/base.py\", line 921, in _map_values\n",
      "    return algorithms.map_array(arr, mapper, na_action=na_action, convert=convert)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/core/algorithms.py\", line 1743, in map_array\n",
      "    return lib.map_infer(values, mapper, convert=convert)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"lib.pyx\", line 2972, in pandas._libs.lib.map_infer\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py\", line 78, in <lambda>\n",
      "    msa_df.sequence = msa_df.sequence.apply(lambda x: x.replace(\".\",\"-\")).apply(lambda x: ''.join([aa.upper() for aa in x]))\n",
      "                                                                                                   ^^^^^^^^^^\n",
      "KeyboardInterrupt\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=5, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A0A2Z5U3Z0_9INFA_Wu_2014\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 565\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 616, in <module>\n",
      "    main(args)\n",
      "  File \"/Users/admin/Documents/Work/2022 Harvard/ProteinGym/notebooks/../proteingym/baselines/esm/compute_fitness.py\", line 424, in main\n",
      "    df = pd.read_csv(args.dms_input)\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/io/parsers/readers.py\", line 1026, in read_csv\n",
      "    return _read(filepath_or_buffer, kwds)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/io/parsers/readers.py\", line 620, in _read\n",
      "    parser = TextFileReader(filepath_or_buffer, **kwds)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/io/parsers/readers.py\", line 1620, in __init__\n",
      "    self._engine = self._make_engine(f, self.engine)\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/io/parsers/readers.py\", line 1880, in _make_engine\n",
      "    self.handles = get_handle(\n",
      "                   ^^^^^^^^^^^\n",
      "  File \"/Users/admin/mambaforge/envs/proteingym/lib/python3.12/site-packages/pandas/io/common.py\", line 873, in get_handle\n",
      "    handle = open(\n",
      "             ^^^^^\n",
      "FileNotFoundError: [Errno 2] No such file or directory: 'per_wt_seq/A0A2Z5U3Z0_9INFA_Wu_2014.csv'\n",
      "Arguments: Namespace(model_type=['MSA_transformer'], model_location=['esm_msa1b_t12_100M_UR50S'], sequence=None, dms_input=PosixPath('per_wt_seq'), dms_index=6, dms_mapping='../reference_files/DMS_substitutions.csv', mutation_col='mutant', dms_output=PosixPath('model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl'), offset_idx=1, scoring_strategy='wt-pseudo-ppl', msa_path=PosixPath('../data/DMS_msa_files'), msa_sampling_strategy='sequence-reweighting', msa_samples=2, msa_weights_folder='../data/DMS_msa_weights/', seeds=[1], filter_msa=False, hhfilter_min_cov=75, hhfilter_max_seq_id=90, hhfilter_min_seq_id=0, path_to_hhfilter='/n/groups/marks/software/hhsuite/hhsuite-3.3.0', scoring_window='overlapping', overwrite_prior_scores=False, device='mps', nogpu=False)\n",
      "Compute scores for DMS: A4_HUMAN_Seuma_2022\n",
      "Note: pseudo-ppl not normalizing by WT\n",
      "Chopping sequence by start 1 and end 770\n",
      "tmp df head:                                      mutated_sequence  DMS_score\n",
      "0  MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMN...          0\n",
      "df shape: (1, 2)\n",
      "Starting model scoring\n",
      "Alphabet for esm_msa1b_t12_100M_UR50S: alphabet.prepend_bos=True, alphabet.append_eos=False\n",
      "Sampling sequences from MSA with strategy: sequence-reweighting\n",
      "Proportion of sequences dropped due to fraction of gaps: 0.0%\n",
      "Proportion of non-focus columns removed: 0.0%\n",
      "/Users/admin/Documents/Work/2022 Harvard/ProteinGym/proteingym/utils/msa_utils.py:100: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
      "  self.seq_name_to_sequence[msa_df.index[seq_idx]] = msa_df.sequence[seq_idx]\n",
      "Encoding sequences\n",
      "One-hot encoded sequences shape:(5230, 770, 20)\n",
      "Loaded sequence weights from disk\n",
      "Neff = 62.3350977696187\n",
      "Data Shape = (5230, 770, 20)\n",
      "Neff: 62.3350977696187\n",
      "Name of focus_seq: >A4_HUMAN/1-770\n",
      "Check sum weights MSA: 1.0\n",
      "First 10 elements of sampled MSA: \n",
      "[('>A4_HUMAN/1-770', 'MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN'), ('>UniRef100_UPI00155F6792/14-753', '------IVAPALAGYIEALAANAGTGFAVAEPQVAMFCGKLNMHVNIQTGNWEPDTSGTKSCLGTKEAVLHYCQEMYPELQITNVVEANQPMRIENWCKKGKKQCKDHAHIVVPYKCLVGEFVSDVLLVPEKCKFFHKERMDVCESHQHWHGVAKEACSSDLMILHSYGMLLPCGIDNFHGTEYVCCPQNRLGDEPESPAQEEEEEEGRDDDEEEDDDDEEEED----FRDEGEEGRDEPQEEQEEEEEDLDYVYEDEDE--EDYRDTEERKTTQDRSPADKESQDLKAVCSLDAETGPCRAMMPRWYFDLDARRCVRFVYGGCAGNRNNFESEEYCMAVCKHL-------------------I-PPPPQPTDDVDVYFETPADENEHARFQRAKEQLEVRHRNRMERVRKEWEEAERQAKNLPKAERQTLIQHFQTMVESLEQEAASEKQQLVETHLARVEAMLNDRRRLALENYLAALQADPPRPHRILQALKRYVRAENKDRQHTFRHYQHLLAVDPEKAEQMKSQVMTHLRVIEERMNQSLSLLYKVPYVAEEIQGEIDVLLQEQRADMDQFVSS-LSESQPDVT----VSSEE-----SDELPAVE---GKPYRPVQVKTMGSQPELEGSGTAGLDGLIGAEERAINSKNKMSENVVIDETLDVKVIFNAERVSGVDEDLVSYRPL-GEFSFSSNALIGLLVIAVAIATVIVISLVMLRKRQYGTISHGIVEVDPMLTPEERHLSKMQNHGYENPTYKYLEQMQ-')]\n",
      "length of first sequence in MSA: 770\n",
      "Batch sizes: torch.Size([1, 2, 771])\n",
      "tmp MSAT logits shape:  torch.Size([1, 2, 771, 33])\n",
      "token probs before unsqueeze:  torch.Size([1, 771, 33])\n",
      "tmp token probs shape:  torch.Size([1, 771, 33])\n",
      "length between indices: 769\n",
      "Ignoring offset_idx because we've already chopped the WT_sequence to the correct length\n",
      "100%|█████████████████████████████████████| 770/770 [00:00<00:00, 251566.76it/s]\n"
     ]
    }
   ],
   "source": [
    "# Using wt-marginals pseudo-ppl (experimental)\n",
    "# Use per-sequence mapping file\n",
    "# for i in range(0, 188):\n",
    "    # command = f\"../proteingym/baselines/esm/compute_fitness.py \\\n",
    "    #     --model-location esm_msa1b_t12_100M_UR50S \\\n",
    "    #     --model_type MSA_transformer \\\n",
    "    #     --dms_index {i} \\\n",
    "    #     --dms_mapping ../reference_files/DMS_substitutions.csv \\\n",
    "    #     --dms-input ./per_wt_seq/ \\\n",
    "    #     --dms-output ./model_scores_out/esm_msa1b_t12_100M_UR50S_wt_pseudo_ppl/ \\\n",
    "    #     --scoring-strategy wt-pseudo-ppl \\\n",
    "    #     --scoring-window overlapping \\\n",
    "    #     --msa-path ../data/DMS_msa_files/ \\\n",
    "    #     --msa-weights-folder ../data/DMS_msa_weights/ \\\n",
    "    #     --msa-samples 2 \\\n",
    "    #     --device mps\"  # Should try use mps\n",
    "    # ! python {command}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Later"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Start with ESM, ProGen for example (easiest two?) and get scores for all WT in the substitutions doc\n",
    "# - use indels scripts, maybe add scripts for unnormalized scores?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Related: Recalculate Spearmans for only singles (need O2 access)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODOs:\n",
    "# - Recalculate Spearmans just for singles to ignore \"flukes\": inflated performance by just separating multiples from singles\n",
    "# Find a way to extend this to alignments\n",
    "# - Difficult: Rescore all Rocklin assays in their sequence context (to get full sequence likelihoods)\n",
    "# - - Although this is already showing a fine correlation just by looking at the tested constructs \n",
    "#       -> and plus these have high spearmans anyway so it's not as if the model was broken here\n",
    "# "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
